backend/src/v4/middlewares/checkPermissions.js

```diff
@@ -124,22 +124,29 @@ function checkMultiplePermissions(permsRequest) {
 			models.push(...req.query.models.split(","));
 		}
 
-		validateUserSession(req, res).then(() => {
+		// PATCH Request
+		if(!models.length && req.body.length) {
+			models.push(...req.body.flatMap((entry)=> entry?.model || []));
+		}
+		try {
+			await validateUserSession(req, res);
+
 			const username = req.session.user.username;
 			const account = req.params.account;
 
-			const promises = [];
+			if(!models.length) {
+				throw responseCodes.INVALID_ARGUMENTS;
+			}
 
-			models.forEach((model) => {
-				const permissionCheckPromise = checkPermissionsHelper(username, account, null, model, permsRequest, getPermissionsAdapter);
-				promises.push(permissionCheckPromise);
-			});
+			const permRes = await Promise.all(models.map((model) => {
+				return checkPermissionsHelper(username, account, null, model, permsRequest, getPermissionsAdapter);
+			}));
 
-			return Promise.all(promises);
-		}).then(validatePermissions.bind(null, next))
-			.catch(err => {
-				next(err);
-			});
+			await validatePermissions(next, permRes);
```

Comment:
not related to this pr but the other place validatePermissions is called (line 106) is passing next as second param... not sure where its called and if its affecting anything just wanted to point it out 😆

Reply:
yea L106 is a old javascript syntax , you'll notice that it's calling the [`bind` function](https://www.w3schools.com/js/js_function_bind.asp) on `validatePermissions` instead of calling validation permissions directly there. Long story short the concept of `this` in super confusing in javascript and before ES6 this was the way to get a function called in the context of something else.

So they are actually doing the same thing for the most part and not a bug on either side.